### DISCUSSION TITLE
Opening diagrams from github on a self-hosted draw.io installation

### DISCUSSION THREAD

**Discussion by fabian-xentral** [jgraph/drawio#4775] [Q&A]
Hey folks,

in order to add custom plugins I've taken the route of self-hosting a draw.io installation on Github Pages.

That works like a charm, but I seem to have lost the ability to open diagrams form github.
I keep getting a "Unknown error" message and see in the console it fails to load

https://[mySite.xyz]/src/main/webapp/github2?getState=1

I've also tried to change this URL to https://[mySite.xyz]/github2?getState=1 to resemble the URL the app.diagrams.net version seems to load, but that also doesn't work.

Can anyone lend me a hand here?

Thank you so much & Best Regards
Fabian

**Comment by davidjgraph** [comment]
Authenication to third-party services has to go via your server, how did you did that previously?

**Reply by fabian-xentral** [reply]
Hi @davidjgraph , 

thank you so much for your quick reply !!

in fact I just came across the https://github.com/jgraph/drawio-github repo. Looks like that might help me achieve what I'm looking for.

We had not yet set up any authentication server-side yet... is this still a prerequisite in conjunction with the drawio-github repo? If so, can you give me any hints that might set me on the right track to research further myself ?

Thank you very much & BR 
Fabian

**Reply by fabian-xentral** [reply]
Hi @davidjgraph , is there any way you could share your github2 app? 
i've been playing around with a node.js app to perform the authentication against GitHub, but haven't been able to hook things up correctly yet.
Thank you & Beset Regards
Fabian

**Comment by fabian-xentral** [comment]
Hi @davidjgraph ,

I'm still struggling quite heavily with two things:

1. I have successfully forked the drawio project and enhanced it with a custom sidebar component. This works nicely until I clear browser cache or move to an incognito window which will break the plugin again.
2. The other issue is that we're not able to get GitHub Auth to work on the customized, self hosted drawio.

Is there any way you can lend us a hand here? Happy to look into consulting services as well - we just really want to get going with this.

Thank you so much & Best Regards
Fabian
